### ISSUE TITLE
Julia: confusion when Julia code is used in a non-standard string literal

### ISSUE THREAD

**Issue by phipsgabler** [codemirror/CodeMirror#4682]
The parser seems to get confused if you use Julia code inside a multi-line string literal with a non-standard prefix. I found this behaviour on [GitHub](https://github.com/wdebeaum/PEG.jl/blob/master/src/PEG.jl), but it's the same in Jupyter Notebook 5.0.0. I've reduced it to:

```julia
doc"""
@rule grammar = "using PEG\n" & rule[*]
string = r"\"(\\.|[^\"])*\""
julia_function = # left as an exercise ;)
"""

function bla(x)
    no code here
end
```

The issue seems specifically have to do with strings inside the literal. It sometimes (in Jupyter and on the CodeMirror homepage) works correctly if the `doc` prefix at the beginning is removed.

**Comment by marijnh** [comment]
@pabloferz or @malmaud, want to take a look at this?

**Comment by pabloferz** [comment]
Fix at #4684

**Comment by marijnh** [comment]
Fixed via #4684
